### ISSUE TITLE
Infinite recursion when fetch with relative path in load function

### ISSUE THREAD

**Issue by jasonlyu123** [sveltejs/kit#2555]
### Describe the bug

This happens when the `paths.base` option is specified. And the load function in the `__layout.svelte` uses the `loadInput.fetch` to load external resources with a relative path (relative to root). But not when it's a full url with a hostname. 

Because the SSR server presumes the route is not external it started another response cycle. Then it would hit the `__layout.svelte` again. Although it also happens if the fetching target is not external. But that's expected. Maybe we can have a similar check like the redirect loop detection. But I think it's also ok without it.

The external url check here probably has to check if the `paths.base` is specified and the url is external because of that
https://github.com/sveltejs/kit/blob/59badb515755a9294365e8084f9beefa10a90773/packages/kit/src/runtime/server/page/load_node.js#L126

### Reproduction

https://github.com/jasonlyu123/kit-ssr-fetch-infinite-recursion-example

### Logs

```shell
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
```

### System Info

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (16) x64 AMD Ryzen 7 4800HS with Radeon Graphics
    Memory: 2.74 GB / 15.42 GB
  Binaries:
    Node: 16.2.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.13.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1023.0), Chromium (94.0.992.38)
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    @sveltejs/adapter-node: ^1.0.0-next.53 => 1.0.0-next.53
    @sveltejs/kit: next => 1.0.0-next.179
    svelte: ^3.42.6 => 3.43.1
```

### Severity

serious, but I can work around it

**Comment by jasonlyu123** [comment]
Ah, I feel dumb. the relative url request shouldn't work. So this issue should be a feature request for infinite recursion detection. At least there's is a friendly error instead of the entire app crash.

**Comment by am283721** [comment]
I also ran into the _layout load function being called recursively ad infinitum when trying to fetch data from an endpoint.

For me it was occurring both in dev and when trying to build for production. And it seemed to only occur when reloading a non-root page that had a dynamic parameter (navigating to the page was fine as _layout was already mounted at that point). Using an absolute path fixed the issue in dev, but not when attempting to build.

I realized that the path I was passing to fetch didn't start with a forward slash and adding it fixed the issue for me.

i.e. `await fetch('/header_resources.json')` rather than `await fetch('header_resources.json')`

This doesn't help narrow down the root cause at all, since we all seem to have different configurations, but figured I'd add my experience to the mix.
